bundles/binding/org.openhab.binding.tacmi/src/main/java/org/openhab/binding/tacmi/internal/message/AnalogValue.java

```diff
@@ -26,21 +26,24 @@
 
     private static Logger logger = LoggerFactory.getLogger(AnalogValue.class);
 
+    // factors for floating point conversion
+    private static final int[] factors = { 0, /* 1 - 5 */ 1, 0, 0, 0, 0, /* 6 - 10 */ 0, 0, 0, 0, 1, /* 11 - 15 */ 1, 0,
+            0, 0, 0, /* 16 - 20 */ 0, 0, 0, 0, 0, /* 21 */ 0 };
+
     /**
      * Create new AnalogValue with specified value and type
      */
     public AnalogValue(int rawValue, int type) {
-        switch (type) {
-            case 1:
-                measureType = TACmiMeasureType.TEMPERATURE;
-                value = new BigDecimal(rawValue).movePointLeft(1);
-                break;
-            case 4:
-                measureType = TACmiMeasureType.SECONDS;
-                value = new BigDecimal(rawValue);
-                break;
-            default:
-                measureType = TACmiMeasureType.UNSUPPORTED;
+        if (type < 1 || type > 21) {
+            measureType = TACmiMeasureType.UNSUPPORTED;
+            value = new BigDecimal(rawValue);
+            logger.warn("Got unsupported measure type {}, value is {}", type, rawValue);
+        } else {
+            measureType = TACmiMeasureType.fromInt(type);
+            value = new BigDecimal(rawValue).movePointLeft(factors[type]);
+            logger.debug("Got measure: type {}, raw value {}, converted: {}, {}", type, rawValue, measureType.name(),
+                    value);
         }
```

Comment:
Better yet, don't range check here at all, just try/catch and let the TACmiMeasureType enum reject unsupported types by throwing the IAE.